src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java

```diff
@@ -171,15 +172,33 @@ public void migrateData()
   @Override
   public boolean configure(StaplerRequest staplerRequest, JSONObject json) throws FormException
   {
+    
+    // Keep a reference to the current configuration to revert back in case of an exception.
+    LogstashIndexer<?> previousIndexer = logstashIndexer;
```

Comment:
I think we don't need this variable, We already have activeIndexer which holds the active configuration and can revert to this.